app/api/config.py

```diff
@@ -0,0 +1,33 @@
+import os
+import socket
+from functools import lru_cache
+
+from pydantic_settings import BaseSettings
+
+USE_CACHED_SETTINGS = os.getenv("USE_CACHED_SETTINGS", "true").lower() == "true"
+
+
+class Settings(BaseSettings):
+    placeholder: str = "This is a placeholder setting."
+    hostname: str = socket.gethostname()
+    model_name: str = "roberta-base"
+    path_to_model_checkpoint: str = "models/AlignScore-base.ckpt"
+    device: str = "cpu"
+
+
+@lru_cache
+def get_cached_settings():
```

Comment:
Can we avoid this pattern if possible, please? We should be creating a single Settings in the lifespan and then pass it down to things that need it. We already have a ticket to remove this singleton in the main repo